### ISSUE TITLE
Use Promises

### ISSUE THREAD

**Issue by RangerMauve** [CloudBoost/JavaScriptSDK#37]
Objects with a "success" and "error" are cool, but Promises are the de facto standard for representing something that will eventually succeed or fail in JS.

**Comment by RangerMauve** [comment]
Seems you actually have promises somewhere in there. This must just be a documentation issue.

**Comment by nawazdhandala** [comment]
Yes, We already have it implemented. We just need to document that feature out. Thank you for your feedback. :)
